src/scheduler.py

```diff
@@ -69,60 +82,98 @@ def parse_args():
         
     args = parse_args()
 
-    cron = CronTab(user='wander')
-    for job in cron:
-        print(job)
+    cron = CronTab(user='arsen')
 
-    output_file = '~/Desktop/output.txt'
+    output_file = '/tmp/cron.lst'
     if args.output is not None:
         output_file = args.output
+
     output_file = output_file + ' 2>&1' # send both stdout and stderr to our output file
 
-    duration = datetime.timedelta(seconds=3600) # in seconds
-    if args.span is not None:
-        duration = datetime.timedelta(seconds=args.span)
-    
-    deadline = datetime.datetime.now() + datetime.timedelta(days=7)
-    if args.deadline is not None:
-        deadline = datetime.datetime.now() + datetime.timedelta(seconds=args.deadline)
-
-    delay = datetime.datetime.now()
-    if args.delay is not None:
-        delay = datetime.datetime.now() + datetime.timedelta(seconds=args.delay)
-
-    model = OfflineModel('NL')
-    task = task.Task(duration, deadline, delay)
-    new_start = model.process_concurrently([task])[0]
-
-    print(new_start)
-
-    if args.repeat is not None:
-        # we have a repeating job, schedule using `cron`
-
-        # extra parentheses around the strings to get a multiline string without whitespace
-        # see: https://stackoverflow.com/a/10660443
-        item = CronItem.from_line((f'{args.repeat} python {os.path.abspath(args.job)} >> '
-                                    f'{output_file}'), cron=cron)
-        cron.append(item)
-        cron.write()
-        print('scheduled repeating job')
-    elif args.at is not None:
-        # we have a one-off job, schedule using `at`
-        format = '%Y%m%d%H%M' # at's format:[[CC]YY]MMDDhhmm
-        if args.green is False:
-            time_str = new_start.strftime(format) 
-        else:
-            time_str = delay.strftime(format)
-        print(time_str)
-        at_options = ""
-        at_time = args.at
-        if args.t is True:
-            at_time = time_str
-            at_options = '-t'
-        cmd = (f'python {os.path.abspath(args.job)} >> {output_file} '
-                f'| at {at_options} {at_time} >> /dev/null') #2>&1')
-        print(cmd)
-        subprocess.run(cmd, shell=True) # TODO: remove shell=True
-        print('scheduled one-off job')
+    isExist = os.path.exists(output_file)
+    if not isExist:
+        # Create a new directory because it does not exist
+        os.makedirs(output_file)
```

Comment:
This is unnecessary: we use `>>` to write to the output, which simply creates the file if it doesn't exist

Reply:
Good point, removed